### PR TITLE
[CPU] Limit f16 in Accuracy mode by hardware support

### PR DESCRIPTION
**Pull request by nshchego**
### Details:
 - *Restrict usage of f16 in transformations pipeline if it's not supported by hardware for ACCURACY MODE*

### Tickets:
 - *145051*

### CODE REVIEW

**Comment by nshchego** [comment]
@maxnick could you please take a look?

File: src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/classes/undefined_et.cpp [hunk 1]
```diff
@@ -135,6 +136,11 @@ fill_data(tensor->data<ov::element_type_traits<P>::value_type>(), S, L); break;
 }
 
 TEST_P(UndefinedEtSubgraphTest, CompareWithRefs) {
+    if (m_data_et == element::f16 && !ov::intel_cpu::hasHardwareSupport(m_data_et)) {
+        abs_threshold = 1.f;
+        rel_threshold = 0.1f;
+    }
+
```
Comment (maxnick) [inline src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/classes/undefined_et.cpp:143]: Usually we define the thresholds in the `SetUp` method

**Review by maxnick — CHANGES_REQUESTED** [review]

**Comment by maxnick** [comment]
General comment. In the PR only the convolution precision selection logic has been modified, but what about all the other operations that rely on oneDNN?

**Comment by nshchego** [comment]
> General comment. In the PR only the convolution precision selection logic has been modified, but what about all the other operations that rely on oneDNN?

The changes in conv fix the original issue from the ticket. The changes in Transformations pipeline is a hot fix for other possible issues.

File: src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/classes/undefined_et.cpp (hunk 1 above)
Comment (nshchego) [reply src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/classes/undefined_et.cpp]: Moved

**Review by maxnick — APPROVED** [review]

**Pull request merged**